kong/status/ready.lua

```diff
@@ -0,0 +1,62 @@
+local declarative = require "kong.db.declarative"
+local constants = require "kong.constants"
+
+local tonumber = tonumber
+local kong = kong
+
+local get_current_hash = declarative.get_current_hash
+
+
+local worker_count = ngx.worker.count()
+local kong_shm = ngx.shared.kong
+local is_traditional = kong.configuration.database ~= "off"
+
+local DECLARATIVE_PLUGINS_REBUILD_COUNT_KEY = 
+                                constants.DECLARATIVE_PLUGINS_REBUILD_COUNT_KEY
+local DECLARATIVE_ROUTERS_REBUILD_COUNT_KEY =
+                                constants.DECLARATIVE_ROUTERS_REBUILD_COUNT_KEY
+local DECLARATIVE_EMPTY_CONFIG_HASH = constants.DECLARATIVE_EMPTY_CONFIG_HASH
+
+local function is_ready()
+
+  local ok = kong.db:connect()
+  if not ok then
+    return false
+  end
+
+  if is_traditional then
+    kong.db:close() -- ignore ERRs
```

Comment:
why?!!!!

Reply:
> why?!!!!

Because previous merged code in `kong/api/routes/health.lua` also do so